frontend-vanilla.js

```diff
@@ -0,0 +1,182 @@
+        "@typescript-eslint/no-shadow": "error",
+        "no-trailing-spaces": [
+            "warn",
+            {
+                "skipBlankLines": true
+            }
+        ],
+        "no-void": ["error", { "allowAsStatement": true }],
+        "no-unused-vars": "off",
+        "@typescript-eslint/no-unused-vars": ["error", {
+            "varsIgnorePattern": "^_",
+            "ignoreRestSiblings": true
+        }],
+        "padding-line-between-statements": [
+            "error",
+            {
+                "blankLine": "always",
+                "prev": "*",
+                "next": "return"
+            },
+            {
+                "blankLine": "always",
+                "prev": [
+                    "multiline-const",
+                    "multiline-let",
+                    "multiline-var",
+                    "multiline-expression",
+                    "multiline-block-like"
+                ],
+                "next": "*"
+            },
+            {
+                "blankLine": "always",
+                "prev": ["const", "let", "var", "expression", "multiline-block-like"],
+                "next": ["export"]
+            },
+            {
+                "blankLine": "always",
+                "prev": "directive",
+                "next": "*"
+            },
+            {
+                "blankLine": "any",
+                "prev": "directive",
+                "next": "directive"
+            },
+            {
+                "blankLine": "always",
+                "prev": "block-like",
+                "next": "*"
+            },
+            {
+                "blankLine": "always",
+                "prev": "expression",
+                "next": "*"
+            },
+            {
+                "blankLine": "always",
+                "prev": ["case", "default"],
+                "next": "*"
+            }
+        ],
+        "prefer-destructuring": [
+            "error",
+            {
+                "object": true,
+                "array": false
+            }
+        ],
+        "prefer-const": ["error"],
+        "prefer-rest-params": ["error"],
+        "prefer-spread": ["off"],
+        "prettier/prettier": [
+            "warn",
+            {
+                "semi": true,
+                "trailingComma": "all",
+                "singleQuote": true,
+                "printWidth": 90,
+                "tabWidth": 2,
+                "endOfLine": "auto",
+                "bracketSpacing": true
+            }
+        ],
+        "@typescript-eslint/no-misused-promises": [
+            "error",
+            {
+                "checksVoidReturn": {
+                    //detects false positives
+                    properties: false
+                }
+            }
+        ],
+        "@typescript-eslint/explicit-function-return-type": ["error", {
+            "allowExpressions": true
+        }],
+        "@typescript-eslint/consistent-type-definitions": ["error", "type"],
```

Comment:
I think it's a good practice to use either types or interfaces (personally prefer types, but this rule can be changed or turned off if preferences differ or if the project is a dependency of another that relies on a specific type definition style)

Reply:
🤷🏻‍♂️ I kinda use interface for complex types and types when I need one line wrappers